ci/infra/testrunner/tests/test_deployments.py

```diff
@@ -1,21 +1,53 @@
+import pytest
+import json
+import requests
+
 from skuba import Skuba
 from kubectl import Kubectl
-import pytest
-import time
-from timeout_decorator import timeout
+
 
 def test_nginx_deployment(setup, skuba, kubectl):
+
+
     skuba.node_join(role="worker", nr=0)
     skuba.node_join(role="worker", nr=1)
     masters = skuba.num_of_nodes("master")
     workers = skuba.num_of_nodes("worker")
     assert masters == 1
     assert workers == 2
-    kubectl.create_deployment("nginx", "nginx:stable-alpine")
-    kubectl.scale_deployment("nginx", workers)
-    kubectl.expose_deployment("nginx", 80)
-    kubectl.wait_deployment("nginx", 3)
-    assert kubectl.count_available_replicas("nginx") == workers
-    result = kubectl.test_service("nginx")
-    assert "Welcome to nginx" in result
+    kubectl.run_kubectl("create deployment nginx --image=nginx:stable-alpine")
+    kubectl.run_kubectl("scale deployment nginx --replicas={replicas}".format(replicas=workers))
+    kubectl.run_kubectl("expose deployment nginx --port=80 --type=NodePort")
+    kubectl.run_kubectl("wait --for=condition=available deploy/nginx --timeout={timeout}m".format(timeout=3))
+    deployment = kubectl.run_kubectl("get deployment/nginx -o json")
+    availableReplicas = readyReplicas = updatedReplicas = replicas = 0
+    try:
+        obj = json.loads(deployment)
+        availableReplicas = int(obj["status"]["availableReplicas"])
+        readyReplicas = int(obj["status"]["readyReplicas"])
+        updatedReplicas = int(obj["status"]["updatedReplicas"])
+        replicas = int(obj["status"]["replicas"])
+    except ex as Exception:
+        raise ex
+
+    assert replicas == workers
+    assert availableReplicas == workers
+    assert readyReplicas == workers
+    assert updatedReplicas == workers
+
+    service = kubectl.run_kubectl("get service/nginx -o json")
+    nodePort = 0
+    try:
+        obj = json.loads(service)
+        nodePort = int(obj["spec"]["ports"][0]["nodePort"])
+    except ex as Exception:
+        raise ex
+    
+    wrk_idx = 0
+    ip_addresses = skuba.platform.get_nodes_ipaddrs("worker")
+    if wrk_idx >= len(ip_addresses):
+        raise ValueError("Node worker-{} not deployed".format(wrk_idx))
```

Comment:
Why using `rise` instead of an `assert? In any case,  this check should be done in the `deployment` fixture.